### PR TITLE
Firebase authentication and private routing [Issues #18 and #76]

### LINKED ISSUE #18: Firebase authentication and mysql
Firebase for authentication with backend verification.
Pretty much all of it explained in this article. 
https://dev.to/emeka/securing-your-express-node-js-api-with-firebase-auth-4b5f

Mysql is then used for the content db. 
Alternatively we could use what they call Realtime database, which is more than a db. Built-in Websockets and I believe hardbound to the authentication system, afaik 🤔

### PR DESCRIPTION
**Pull request by cecastosic**
resolves #18 #76 
not finished - PR made so the whole team can work together on issues 
frontend part done

### CODE REVIEW

**Comment by senner007** [comment]
input-login and login should be in the same folder, imo

File: src/client/containers/login-page/login-page.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    <main>
+      <Login
+        title="Login"
+        errorNotifications={errorNotifications}
+        errorMessages={errorMessage}
+        handleChange={(event) => {
+          if (event.target.type === 'email') setEmail(event.target.value);
+          if (event.target.type === 'password') setPassword(event.target.value);
+        }}
+        loginFunc={async (event) => {
```
Comment (senner007) [inline src/client/containers/login-page/login-page.js]: move logic out of view please

File: src/client/containers/login-page/login-page.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [errorNotifications, setErrorNotifications] = useState(null);
+
+  const history = useHistory();
+
+  return (
+    <main>
+      <Login
```
Comment (senner007) [inline src/client/containers/login-page/login-page.js]: A bit too generic here. I think you should create an email and a password component.

**Comment by senner007** [comment]
Plenty of single responsibility thinking here
...makes the 🐱‍👤 coder envious ...

File: src/client/App.js [hunk 3]
```diff
@@ -1,14 +1,25 @@
...
+        </Route>
+        <Route exact path="/dashboard">
+          <Dashboard />
+        </Route>
         <Route exact path="/">
-          <Home />
+          <FirebaseContext.Provider value={Firebase.init()}>
+            <LoginPage />
+          </FirebaseContext.Provider>
         </Route>
```
Comment (bytewiz) [inline src/client/App.js]: I think I would wrap the entire app in the context provider as you might want to be able to display state about whether you are signed in or not in some of the pages even though they don't require authentication 👍

**Comment by senner008** [comment]
I suggest that this pr. is closed. Then finish the work on the branch and open another one

File: src/client/App.js (hunk 3 above)
Comment (cecastosic) [reply src/client/App.js]: Done

**Comment by cecastosic** [comment]
@bytewiz We added private and public routes: If you are not logged in, it will show login form, if you are logged in, you can see dashboard. 
All changes related to components are there because of showing needed components on dashboard page.

**Comment by zenkaeser** [comment]
Hello guys, Daniel asked me if you have already a Swagger Documentation for your endpoint on this issue if there's any?Thank you

**Comment by bytewiz** [comment]
@sgalkina Only some conflicting files left it seems 👍 Please have a look at it 👍

**Comment by cecastosic** [comment]
Resolved 👍

**Pull request merged**